daemon/container_operations_unix.go

```diff
@@ -625,11 +690,15 @@ func (daemon *Daemon) ConnectToNetwork(container *container.Container, idOrName
 	return nil
 }
 
-func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName string, updateSettings bool) (err error) {
+func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName string, endpointConfig *networktypes.EndpointSettings, updateSettings bool) (err error) {
 	if container.HostConfig.NetworkMode.IsContainer() {
 		return runconfig.ErrConflictSharedNetwork
 	}
 
+	if !container.HostConfig.NetworkMode.IsUserDefined() && networktypes.HasUserDefinedIPAddress(endpointConfig) {
```

Comment:
Thanks @tonistiigi @mavenugo, yes that check is wrong, it needs to check on the network which we are connecting to.

@mavenugo I am already checking for the subnet being configured: https://github.com/docker/docker/pull/19001/files#diff-0f20873a38571444bac38770160648a5R715